modin/experimental/core/execution/dask/implementations/pandas_on_dask/io/io.py

```diff
@@ -0,0 +1,124 @@
+
+    ``ExperimentalPandasOnDaskIO`` inherits some util functions and unmodified IO functions
+    from ``PandasOnDaskIO`` class.
+    """
+
+    build_args = dict(
+        frame_partition_cls=PandasOnDaskDataframePartition,
+        query_compiler_cls=PandasQueryCompiler,
+        frame_cls=PandasOnDaskDataframe,
+        base_io=PandasOnDaskIO,
+    )
+
+    def __make_read(*classes, build_args=build_args):  # noqa: GL08
+        # used to reduce code duplication
+        return type("", (DaskWrapper, *classes), build_args)._read
+
+    read_csv_glob = __make_read(PandasCSVGlobParser, CSVGlobDispatcher)
+
+    read_pickle_distributed = __make_read(
+        PandasPickleExperimentalParser, PickleExperimentalDispatcher
+    )
+
+    read_custom_text = __make_read(
+        CustomTextExperimentalParser, CustomTextExperimentalDispatcher
+    )
+
+    read_sql = __make_read(SQLExperimentalDispatcher)
+
+    del __make_read  # to not pollute class namespace
+
+    @classmethod
+    def to_pickle_distributed(cls, qc, **kwargs):
+        """
+        When `*` in the filename all partitions are written to their own separate file.
+
+        The filenames is determined as follows:
+        - if `*` in the filename then it will be replaced by the increasing sequence 0, 1, 2, …
+        - if `*` is not the filename, then will be used default implementation.
+
+        Examples #1: 4 partitions and input filename="partition*.pkl.gz", then filenames will be:
+        `partition0.pkl.gz`, `partition1.pkl.gz`, `partition2.pkl.gz`, `partition3.pkl.gz`.
+
+        Parameters
+        ----------
+        qc : BaseQueryCompiler
+            The query compiler of the Modin dataframe that we want
+            to run ``to_pickle_distributed`` on.
+        **kwargs : dict
+            Parameters for ``pandas.to_pickle(**kwargs)``.
+        """
+        if not (
+            isinstance(kwargs["filepath_or_buffer"], str)
+            and "*" in kwargs["filepath_or_buffer"]
+        ) or not isinstance(qc, PandasQueryCompiler):
+            warnings.warn("Defaulting to Modin core implementation")
+            return PandasOnDaskIO.to_pickle(qc, **kwargs)
+
+        def func(df, **kw):
+            idx = str(kw["partition_idx"])
+            # dask doesn't make a copy of kwargs on serialization;
+            # so take a copy ourselves, otherwise the error is:
+            #  kwargs["path"] = kwargs.pop("filepath_or_buffer").replace("*", idx)
+            #  KeyError: 'filepath_or_buffer'
+            dask_kwargs = dict(kwargs)
+            dask_kwargs["path"] = dask_kwargs.pop("filepath_or_buffer").replace(
+                "*", idx
+            )
+            df.to_pickle(**dask_kwargs)
+            return pandas.DataFrame()
+
+        result = qc._modin_frame.broadcast_apply_full_axis(
+            1, func, other=None, new_index=[], new_columns=[], enumerate_partitions=True
+        )
+        result.to_pandas()  # wait for completion of all tasks
```

Comment:
should we use wait instead of to_pandas?

Reply:
we can, code is updated to use `wait_partitions`.